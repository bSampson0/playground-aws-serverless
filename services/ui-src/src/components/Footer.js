import React from "react";
import { Container } from "react-bootstrap";
import { helpDeskContact } from "../libs/contacts";
import { HelpSection } from "./HelpSection";

function TopFooter() {
  return (
    <section className="footer-top-container">
      <Container>
        <div className="footer-wrapper footer-top-wrapper">
          <div>
            <img src="/footer/logo-MedicaidGov.svg" alt="Medicaid.gov logo" />
          </div>
          <div className="footer-government-blurb">
            <div className="footer-fed-gov-container">
              <img
                src="/footer/depthealthhumanservices_usa.svg"
                alt="Department of Health and Human Services logo"
              ></img>
            </div>
            <div className="footer-fed-gov-text">
              A federal government website managed and paid for by the U.S.
              Centers for Medicare and Medicaid Services and part of the MACPro
              suite.
            </div>
          </div>
        </div>
      </Container>
    </section>
  );
}

function BottomFooter() {
  return (
    <div className="footer-bottom-container">
      <Container>
        <div className="footer-wrapper">
          <div>
            Email{" "}
            <a
              className="footer-email"
              href={`mailto:${helpDeskContact.email}`}
            >
              {" "}
              {helpDeskContact.email}
            </a>{" "}
            for help or feedback.
          </div>
          <div>7500 Security Boulevard Baltimore, MD 21244</div>
        </div>
      </Container>
    </div>
  );
}

function Footer() {
  return (
    <footer>
      <HelpSection />
      <TopFooter />
      <BottomFooter />
    </footer>
  );
}

export default Footer;
