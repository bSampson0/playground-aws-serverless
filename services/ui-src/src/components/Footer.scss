.footer-top-container,
.footer-bottom-container {
  .header-wrapper,
  .footer-wrapper {
    @extend .ds-u-display--flex;
    @extend .ds-u-align-items--center;
    @extend .ds-u-justify-content--between;
    width: 100%;
  }

  @media only screen and (max-width: $width-lg) {
    .footer-wrapper {
      flex-direction: column;
    }

    .footer-top-wrapper {
      > * {
        margin-top: 40px;
      }
    }
  }
}

.footer-top-container {
  @extend .ds-u-padding-top--0;
  @extend .ds-u-lg-padding-top--2;
  @extend .ds-u-padding-bottom--3;
  background: $color-primary-alt-lightest;
}
.footer-bottom-container {
  @extend .ds-u-padding-y--1;
  background: $color-primary;
  color: $color-white;
  font-size: 14px;
  line-height: 2.1;
  letter-spacing: 0;
}
.footer-government-blurb {
  @extend .ds-u-display--flex;
  @extend .ds-u-align-items--center;
  max-width: 480px;
}
.footer-fed-gov-logo {
  min-width: 80px;
}
.footer-fed-gov-text {
  font-size: 14px;
  line-height: 21px;
  text-align: justify;
  max-width: 720px;
}
.footer-email {
  color: $color-white;
}
