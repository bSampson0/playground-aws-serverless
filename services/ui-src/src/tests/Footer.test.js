import React from "react";
import { render } from "@testing-library/react";
import Footer from "../components/Footer";

describe("Test Footer.js", () => {
  test("Should match snapshot", () => {
    expect(render(<Footer />).asFragment()).toMatchSnapshot();
  });
});
