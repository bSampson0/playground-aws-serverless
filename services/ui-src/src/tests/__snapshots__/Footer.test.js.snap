// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer.js Should match snapshot 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="mb-4 container"
    >
      <h3>
        Do you have questions or need support?
      </h3>
      <div
        class="footer-fed-gov-text"
      >
        For technical questions regarding use of this application, please reach out to 
        <a
          href="mailto:MDCT_Help@cms.hhs.gov"
        >
          MDCT_Help@cms.hhs.gov
        </a>
        . For content-related questions, such as about measure specifications or what information to enter in each field, please reach out to 
        <a
          href="mailto:MACQualityTA@cms.hhs.gov"
        >
          MACQualityTA@cms.hhs.gov
        </a>
      </div>
    </div>
    <section
      class="footer-top-container"
    >
      <div
        class="container"
      >
        <div
          class="footer-wrapper footer-top-wrapper"
        >
          <div>
            <img
              alt="Medicaid.gov logo"
              src="/footer/logo-MedicaidGov.svg"
            />
          </div>
          <div
            class="footer-government-blurb"
          >
            <div
              class="footer-fed-gov-container"
            >
              <img
                alt="Department of Health and Human Services logo"
                src="/footer/depthealthhumanservices_usa.svg"
              />
            </div>
            <div
              class="footer-fed-gov-text"
            >
              A federal government website managed and paid for by the U.S. Centers for Medicare and Medicaid Services and part of the MACPro suite.
            </div>
          </div>
        </div>
      </div>
    </section>
    <div
      class="footer-bottom-container"
    >
      <div
        class="container"
      >
        <div
          class="footer-wrapper"
        >
          <div>
            Email 
            <a
              class="footer-email"
              href="mailto:MDCT_Help@cms.hhs.gov"
            >
               MDCT_Help@cms.hhs.gov
            </a>
             for help or feedback.
          </div>
          <div>
            7500 Security Boulevard Baltimore, MD 21244
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
